### PR TITLE
my code

### PR DESCRIPTION
**Pull request by Hamidreza-Fakharzadeh**

### CODE REVIEW

**Comment by Hamidreza-Fakharzadeh** [comment]
I appreciate it if there is any suggestion.

File: week-3/Homework/mandatory/1-oxygen-levels.js [hunk 1]
```diff
@@ -8,33 +8,43 @@ To be safe, they need to land on the first unamed planet that has Oxygen levels
...
+  let newString = index.slice(0, 4);
+  return newString;
+}
+function funcValue(value) {
+  return value > 19.5 && value < 23.5;
+}
 
-function safeLevels() {
-
+function safeLevels(oxygenLevels) {
```
Comment (AlfioEmanueleFresta) [inline week-3/Homework/mandatory/1-oxygen-levels.js]: Well done!

File: week-3/Homework/mandatory/2-bush-berries.js [hunk 2]
```diff
@@ -9,26 +9,40 @@
...
+  } else {
+    return 'Bush is safe to eat from';
+  }
+}
 
-function bushChecker() {
-
+function bushChecker(paramArr) {
+  console.log(paramArr.filter(funcMap));
+  return paramArr.filter(funcMap);
```
Comment (AlfioEmanueleFresta) [inline week-3/Homework/mandatory/2-bush-berries.js]: The objective here is to determine whether the bush has *any* pink berries. 

Hint: Can you *find* any pink berries in your bush?

File: week-3/Homework/mandatory/3-space-colonies.js [hunk 3]
```diff
@@ -7,26 +7,31 @@
 
   NOTE: don't include any element that is not a "family".
 */
+function funcMap(index) {
+  return index.indexOf(' ') !== -1 && index[0] === 'A';
```
Comment (AlfioEmanueleFresta) [inline week-3/Homework/mandatory/3-space-colonies.js]: Well done! This does the job.

I would encourage you to look into ways you can make your code easier to read - it's a very useful skill when working in a team.

If you Google the specific task you're trying to achieve (e.g. *check if a word is in a string javascript*) you will often find convenient functions that help you do your task, but may bit easier to use or read, than what you have in mind.

As an example, in this case, you could achieve the same result using the well-named functions:
* `includes`, which can tell you if a string contains a certain word, and
* `startsWith`, which can tell you if a string starts with a certain letter

```javascript
function isGoodFamilyName(name) {
    return name.includes("family") && name.startsWith("A");
}
```

**Comment by AlfioEmanueleFresta** [comment]
Exercise 4 can be a bit trickier, because each entry is made up of two elements: the student name, and their attendance.

```javascript
const attendances = [
  ["Ahmed", 8],
  ["Clement", 10],
  ["Elamin", 6],
  ["Adam", 7],
  ["Tayoa", 11],
  ["Nina", 10]
]
```

If you use `filter` or `map` on an array like this, your function will receive an argument which is an array of two elements:

```javascript
attendances.forEach(checkIndividualAttendance);

function checkIndividualAttendance(attendance) {
  // attendance here will be something like: ["Nina", 10]
  // If you're only interested in the name, you can access the first element in the array: 
  let name = attendance[0];
  console.log(name); // This will print "Nina"
  // Now, how would you access the number of classes (10)? 
}
```



```

**Comment by Hamidreza-Fakharzadeh** [comment]
thank you for your attention

**Comment by Hamidreza-Fakharzadeh** [comment]
thank you

**Comment by Hamidreza-Fakharzadeh** [comment]
hi

**Comment by Hamidreza-Fakharzadeh** [comment]
thank you for checking

**Comment by Hamidreza-Fakharzadeh** [comment]
thank you for checking

**Comment by Hamidreza-Fakharzadeh** [comment]
thank you

**Comment by Hamidreza-Fakharzadeh** [comment]
thank you for reviewing

**Comment by Hamidreza-Fakharzadeh** [comment]
thank you for checking my code

**Pull request closed without merging**